### ISSUE TITLE
Allow filtering with a series/list with `.is_in()`

### ISSUE THREAD

**Issue by markfairbanks** [pola-rs/polars#610]
It would be helpful if the `.is_in()` method would work with a series and/or a list.

pandas works with both, but even just a series would be very helpful in polars.

pandas example:
```python
import pandas as pd
import numpy as np

df = pd.DataFrame(
    {'x': np.arange(1, 4),
     'y': ['a', 'b', 'c']}
)

letters = pd.Series(['a', 'b'])

df[df.y.isin(letters)]
#>    x  y
#> 0  1  a
#> 1  2  b
```

Syntax in polars:
```python
df.lazy().filter(col('y').is_in(letters))
```

**Comment by ritchie46** [comment]
What do you mean with a list?

**Comment by markfairbanks** [comment]
In the example above `letters` is a series, but you can also filter using a list of values:

```python
import pandas as pd
import numpy as np

df = pd.DataFrame(
    {'a': np.arange(1, 4),
     'b': ['a', 'b', 'c']}
)

df[df.b.isin(['a', 'b'])]
#>    a  b
#> 0  1  a
#> 1  2  b
```

**Comment by ritchie46** [comment]
> In the example above `letters` is a series, but you can also filter using a list of values:
> 
> ```python
> import pandas as pd
> import numpy as np
> 
> df = pd.DataFrame(
>     {'x': np.arange(1, 4),
>      'y': ['a', 'b', 'c']}
> )
> 
> df[df.y.isin(['a', 'b'])]
> #>    x  y
> #> 0  1  a
> #> 1  2  b
> ```
> 
> Edit: I think a better way to phrase my original request would have been "Allow the use of a series or list inside of `.is_in()`"

Clear! :+1:

**Comment by markfairbanks** [comment]
This one doesn't seem to work for me:

```python
import polars as pl
from polars import col
import numpy as np

df = (
    pl.DataFrame({'x': np.arange(1, 4),
                  'y': ['a', 'b', 'c']})
    .lazy()
)

df.filter(col('y').is_in(['a', 'b'])).collect()
#> Traceback (most recent call last):
#> <ipython-input-1-4efae31cb2c5> in <module>
#> ----> 1 df.filter(col('y').is_in(['a', 'b'])).collect()
#> ~/opt/miniconda3/lib/python3.9/site-packages/polars/lazy/__init__.py in is_in(self, other)
#>    1458         if type(other) is list:
#>    1459             other = lit(Series("", other))
#> -> 1460         return wrap_expr(self._pyexpr.is_in(other))
#>    1461 
#>    1462     def is_between(
#> TypeError: argument 'expr': 'Expr' object cannot be converted to 'PyExpr'
```

**Comment by ritchie46** [comment]
My bad, should be fixed in #610

**Comment by ritchie46** [comment]
> This one still isn't working for me
> 
> ```python
> import polars as pl
> from polars import col
> import numpy as np
> 
> df = (
>     pl.DataFrame({'x': np.arange(1, 4),
>                   'y': ['a', 'b', 'c']})
>     .lazy()
> )
> 
> df.filter(col('y').is_in(['a', 'b'])).collect()
> #> Traceback (most recent call last):
> #> <ipython-input-1-4efae31cb2c5> in <module>
> #> ----> 1 df.filter(col('y').is_in(['a', 'b'])).collect()
> #> ~/opt/miniconda3/envs/phish/lib/python3.9/site-packages/polars/lazy/__init__.py in collect(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, string_cache, no_optimization)
> #>     269             string_cache,
> #>     270         )
> #> --> 271         return wrap_df(ldf.collect())
> #>     272 
> #>     273     def fetch(
> #> RuntimeError: Any(ValueError("Series length should be equal to that of the DataFrame."))
> ```

Will investigate this one.
